### ISSUE TITLE
Feature: accept wildcard when installing a version

### ISSUE THREAD

**Issue by felipecassiors** [sdkman/sdkman-cli#776]
## Description
Accepting a wildcard when installing a version would help users to install the latest major version of one SDK.

## Example
```bash
sdk install java 8.*.hs-adpt
# Then sdk is supposed to install the latest AdoptOpenJDK 8 Hotspot
```

Optionally, by using a flag such as:
```bash
sdk install --wildcard java 8.*.hs-adpt
```

## Workaround
I'm currently doing this to get the same behavior:
```bash
sdk i java "$(sdk ls java | grep -o '8.*.hs-adpt' | awk '{print $NF}')"
```

**Comment by banshee** [comment]
As a new user, * as the operator for this would be strange.  Seems obvious to me that "sdk install java 8.*.hs-adpt" would install every available version, not the latest.  You'd probably want another flag, so ```sdk install java 8.*.hs-adpt``` would install everything, and ```sdk install java --only-latest 8.*.hs-adpt``` would just be the latest.  (And you'd need a strict definition of "latest" - is that always something you can discover with version identifiers, and is that _guaranteed_ to be true across every project that wants to be in sdkman?)

I can see actually wanting 8.* to get everything, particularly if you're a library author and you'd like to run tests against many jvm versions.

What about doing something like using maven-style notation instead? 
https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN402  So ```sdk install java [8,9)``` would install the latest version of 8.

**Comment by felipecassiors** [comment]
> I can see actually wanting 8.* to get everything, particularly if you're a library author and you'd like to run tests against many jvm versions.

That's true, you're right. But when I run `sdk list java` it only returns a single version for each of the major versions, that's why I thought the glob would not expand to more than one occurrence. But you're right, this is not obvious. 

My goal with this feature is to be able to install the latest version of a given major version of the candidate.

> What about doing something like using maven-style notation instead?
> https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN402 So `sdk install java [8,9)` would install the latest version of 8.

That sounds very cool and user friendly, but remember that for some candidates, such as Java, there are different flavors of a major version. For example I want to install latest Java 8 - AdoptOpenJDK Hostpot.

**Comment by banshee** [comment]
"That sounds very cool and user friendly, but remember that for some candidates, such as Java, there are different flavors of a major version. For example I want to install latest Java 8 - AdoptOpenJDK Hostpot."  Yes, maven-style here seems like a bad fit since it can't be made to work with all the identifiers that sdkman uses.

**Comment by helpermethod** [comment]
The question IMHO is if this is to be solved client-side or server-side. E.g. the server could support a version like `8.latest.hs-adpt`.

**Comment by felipecassiors** [comment]
We could invent a new kind of identifier: major identifiers:

```sh-session
$ sdk list java
================================================================================
Available Java Versions
================================================================================
 Vendor        | Use | Version      | Dist    | Status    | Major Identifier | Identifier
--------------------------------------------------------------------------------
 AdoptOpenJDK  |     | 8.0.262.hs   | adpt    |           | 8.hs-adpt        |  8.0.262.hs-adpt

$ sdk install java 8.hs-adpt

**Comment by bruno-lopes** [comment]
> We could have more than one identifier so:
> 
> ```
> $ sdk list java
> ================================================================================
> Available Java Versions
> ================================================================================
>  Vendor        | Use | Version      | Dist    | Status    | Identifier
> --------------------------------------------------------------------------------
>  AdoptOpenJDK  |     | 8.0.262.hs   | adpt    |           |  8.hs-adpt, 8.0.262.hs-adpt
> 
> $ sdk install java 8.hs-adpt
> ```

I think that is a very good solution!

**Comment by khmarbaise** [comment]
> https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN402

Do not use that as a reference because it's in the majority Maven 2 based. So long out of date.

**Comment by khmarbaise** [comment]
@bruno-lopes That suggestion looks easy and very concise. That would mean I could also use such thing in my `.sdkmanrc` file.. 
```
# Enable auto-env through the sdkman_auto_env config
# Add key=value pairs of SDKs to use below
java=8.hs-adpt
```

**Comment by alexgagnon** [comment]
I'm not sure of the underlying mechanics, but to me having it on the client side would make more sense, because it would also be useful to also do something like `sdk use java 8` instead of having to type out the full candidate version. My use case is I regularly switch between 8 and 11 (so `sdk env` would work here), but I also have install scripts to bringing down sdkman and multiple SDKs, which as @khmarbaise links can get out of date. I find I don't often care if one of my projects uses 8.0.282.hs-adpt versus 8.0.265.hs-adpt,

Could a wildcard in the `.sdkmanrc` be misleading? I.e. is it taking any version of the SDK already available to the system, or does it go and grab the latest?

Might be outside the scope of this issue, but might it be worth even just being able to create explicit aliases that sdkman can reference? Something like `sdk alias <alias> <candidate>`, where candidate here could use the above wildcard suggestion?

**Comment by alexgagnon** [comment]
Any updates on this? I tried bringing this up in the chat group but I didn't get any responses.
